core/io/ip.cpp

```diff
@@ -179,7 +183,8 @@ IP_Address IP::get_resolve_item_address(ResolverID p_id) const {
 
 	if (resolver->queue[p_id].status != IP::RESOLVER_STATUS_DONE) {
 		ERR_EXPLAIN("Resolve of '" + resolver->queue[p_id].hostname + "'' didn't complete yet.");
-		ERR_FAIL_COND_V(resolver->queue[p_id].status != IP::RESOLVER_STATUS_DONE, IP_Address());
+		resolver->mutex->unlock();
+		return IP_Address();
```

Comment:
I don't even know why this used `ERR_FAIL_COND` in the first place when there is a self explanatory error message already. Either `ERR_PRINT` or `ERR_EXPLAIN` followed by `ERR_FAIL_V` will be fine.